src/trace/cltrace.cpp

```diff
@@ -307,6 +584,26 @@ cl_mem clCreateBuffer(cl_context context, cl_mem_flags flags, size_t size,
     return ret;
 }
 
+cl_mem clCreateSubBuffer(cl_mem buffer, cl_mem_flags flags,
+                         cl_buffer_create_type buffer_create_type,
+                         const void* buffer_create_info, cl_int* errcode_ret) {
+    auto ret = PFN_clCreateSubBuffer(buffer, flags, buffer_create_type,
+                                     buffer_create_info, errcode_ret);
+
+    Call call(oclapi::command::CREATE_SUB_BUFFER);
+    call.record_object_use(buffer);
+    call.record_value(flags);
+    call.record_value(buffer_create_type);
+    call.record_value_out_by_reference(const_cast<void*>(buffer_create_info),
+                                       sizeof(_cl_buffer_region));
```

Comment:
```suggestion
    call.record_value_out_by_reference(const_cast<void*>(buffer_create_info),
                                       sizeof(_cl_buffer_region)); // FIXME this is incorrect
```

The value is an input value, hence why you needed the `const_cast`. The correct type to use depends on buffer_create_type. I'll get this code in commented out as a starting point.